src/app/dim-ui/destiny-symbols/destiny-symbols.tsx

```diff
@@ -0,0 +1,122 @@
+import { D2ManifestDefinitions } from 'app/destiny2/d2-definitions';
+import { t, tl } from 'app/i18next-t';
+import { d2ManifestSelector } from 'app/manifest/selectors';
+import { FontGlyphs } from 'data/d2/d2-font-glyphs';
+import { createSelector } from 'reselect';
+import { conversionTableSelector } from './rich-destiny-text';
+
+type categoryNames = keyof typeof symbols;
+const categoryNameTranslations: { [key in categoryNames]: string } = {
+  abilities: tl('Glyphs.Abilities'),
+  misc: tl('Glyphs.Miscellaneous'),
+  supers: tl('Glyphs.Supers'),
+  weapons: tl('Glyphs.Weapons'),
+};
+
+type SymbolsMap = {
+  name: string;
+  symbols: { glyph: string; name: string; fullName: string }[];
+}[];
+
+const getHardcodedLoc = (glyph: FontGlyphs) => {
+  switch (glyph) {
+    case FontGlyphs.gilded_title:
+      return t('Glyphs.Gilded');
+    case FontGlyphs.hunter_smoke:
+      return t('Glyphs.Smoke');
+    case FontGlyphs.environment_hazard:
+      return t('Glyphs.Misadventure');
+    default:
+  }
+};
+
+const getTableLoc = (defs: D2ManifestDefinitions, tableName: string, hash: number) => {
+  switch (tableName) {
+    case 'Trait':
+      return defs.Trait.get(hash)?.displayProperties?.name;
+    case 'InventoryItem':
+      return defs.InventoryItem.get(hash)?.displayProperties?.name;
+    case 'SandboxPerk':
+      return defs.SandboxPerk.get(hash)?.displayProperties?.name;
+    case 'ActivityMode':
+      return defs.ActivityMode[hash]?.displayProperties?.name;
+  }
+};
+
+const simplifyName = (name: string) => name.toLowerCase().replace(/[^\p{L}]/gu, '');
+
+export const symbolsSelector = createSelector(
+  d2ManifestSelector,
+  conversionTableSelector,
+  (defs, richTextReplacements) => {
+    const list: SymbolsMap = [];
+    if (!defs) {
+      return list;
+    }
+
+    for (const [cat_, catSymbols] of Object.entries(symbols)) {
+      const cat = cat_ as categoryNames;
+      const outCat: SymbolsMap[number] = { name: t(categoryNameTranslations[cat]), symbols: [] };
+      for (const entry of catSymbols) {
+        const hardCodedName = getHardcodedLoc(entry.glyph);
+        const glyph = String.fromCodePoint(entry.glyph);
+        if (hardCodedName) {
+          outCat.symbols.push({
+            glyph,
+            fullName: hardCodedName,
+            name: simplifyName(hardCodedName),
+          });
+          continue;
+        }
+        if ('source' in entry) {
+          const source = entry.source!;
+          const defName = getTableLoc(defs, source.tableName, source.hash);
+          if (defName) {
+            outCat.symbols.push({ glyph, fullName: defName, name: simplifyName(defName) });
+            continue;
+          }
+        }
+        const richTextRepl = richTextReplacements[glyph];
+        if (richTextRepl) {
+          const fullName = richTextRepl.plaintext.slice(1, -1);
+          outCat.symbols.push({ glyph, fullName, name: simplifyName(fullName) });
+          continue;
+        }
```

Comment:
This uses the existing richTextReplacements as a fallback but I'd probably want to roll those into d2ai entirely too.